relion/protocols/protocol_base.py

```diff
@@ -214,59 +230,113 @@ def _defineParams(self, form):
+                          help='A volume mask containing a (soft) mask with '
+                               'the same dimensions as the reference(s), '
+                               'and values between 0 and 1, with 1 being 100% '
+                               'protein and 0 being 100% solvent. The '
+                               'reconstructed reference map will be multiplied '
+                               'by this mask. If no mask is given, a soft '
+                               'spherical mask based on the <radius> of the '
+                               'mask for the experimental images will be '
+                               'applied.\n\n'
+                               'In some cases, for example for non-empty '
+                               'icosahedral viruses, it is also useful to use '
+                               'a second mask. Check _Advaced_ level and '
+                               'select another volume mask')
+            form.addParam('solventMask', PointerParam, pointerClass='VolumeMask',
+                          expertLevel=LEVEL_ADVANCED, allowsNull=True,
+                          label='Second reference mask (optional)',
+                          help='For all white (value 1) pixels in this second '
+                               'mask the corresponding pixels in the '
+                               'reconstructed map are set to the average value '
+                               'of these pixels. Thereby, for example, the '
+                               'higher density inside the virion may be set to '
+                               'a constant. Note that this second mask should '
+                               'have one-values inside the virion and '
+                               'zero-values in the capsid and the solvent '
+                               'areas.')
+            form.addParam('solventFscMask', BooleanParam, default=False,
+                          expertLevel=LEVEL_ADVANCED,
+                          label='Use solvent-flattened FSCs?',
+                          help='If set to Yes, then instead of using '
+                               'unmasked maps to calculate the gold-standard '
+                               'FSCs during refinement, masked half-maps '
+                               'are used and a post-processing-like '
+                               'correction of the FSC curves (with '
+                               'phase-randomisation) is performed every '
+                               'iteration. This only works when a reference '
+                               'mask is provided on the I/O tab. This may '
+                               'yield higher-resolution maps, especially '
+                               'when the mask contains only a relatively '
+                               'small volume inside the box.')
+            # TODO: Check if referenceMask is used in 2D classification protocol in Relion
```

Comment:
yes, I used 2D mask for focused 2D classification. But it was only in continue mode with setting priors